sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -3167,28 +3279,34 @@ pi_result piQueueRelease(pi_queue Queue) {
       // Make sure all commands get executed.
       // Only do so for a healthy queue as otherwise sync may not be valid.
       if (Queue->Healthy) {
-        for (auto &ZeQueue : Queue->ComputeQueueGroup.ZeQueues) {
-          if (ZeQueue)
-            ZE_CALL(zeHostSynchronize, (ZeQueue));
-        }
-        for (auto &ZeQueue : Queue->CopyQueueGroup.ZeQueues) {
-          if (ZeQueue)
-            ZE_CALL(zeHostSynchronize, (ZeQueue));
+        if (UseImmediateCommandLists) {
+          Queue->waitOnAllImmCmdLists();
+        } else {
+          for (auto &ZeQueue : Queue->ComputeQueueGroup.ZeQueues) {
+            if (ZeQueue)
+              ZE_CALL(zeHostSynchronize, (ZeQueue));
+          }
+          for (auto &ZeQueue : Queue->CopyQueueGroup.ZeQueues) {
+            if (ZeQueue)
+              ZE_CALL(zeHostSynchronize, (ZeQueue));
+          }
         }
       }
-
       // Destroy all the fences created associated with this queue.
       for (auto it = Queue->CommandListMap.begin();
            it != Queue->CommandListMap.end(); ++it) {
         // This fence wasn't yet signalled when we polled it for recycling
         // the command-list, so need to release the command-list too.
-        if (it->second.InUse) {
+        // For immediate commandlists we don't need to resset the commandlist
+        // but rely on event cleanup done as part of reset.
+        if (UseImmediateCommandLists || it->second.InUse) {
```

Comment:
The comment is not in sync with the code: do you want or not want to call the "reset"?

Reply:
I have reworded the comment.